### PR TITLE
feat(rtl): optimize the new mixins for smaller bundle, ltr seperation

### PR DESCRIPTION
**Pull request by AmitMY**
#### Short description of what this resolves:
- Mixins were not very efficient, creating 4 properties when there can be 1 most of the time.
- LTR rules override some RTL rules, or RTL auto/initial override RTL itself, fixed
- No direction can only be set for LTR

#### Changes proposed in this pull request:

- Better mixins generation (smaller bundle size)
- Better RTL support
- Choose a direction! `$app-direction: null` is the default, which means all available directions. `$app-direction: ltr` or `$app-direction: rtl` make sure the app only support one direction (reducing bundle from 687kb to 575kb, using everything default)

**Ionic Version**: 3.x

Fixes preview: http://ionic-snapshot-go.appspot.com/ionic2/snapshots/d4t/hhp/chrome_400x800

Closes: https://github.com/ionic-team/ionic/issues/11805 https://github.com/ionic-team/ionic/issues/11914

### CODE REVIEW

File: src/components/action-sheet/action-sheet.ios.scss [hunk 1]
```diff
@@ -97,16 +97,15 @@ $action-sheet-ios-button-cancel-font-weight:            600 !default;
 
 .action-sheet-ios .action-sheet-group {
   @include border-radius($action-sheet-ios-border-radius);
+  @include margin(null, null, $action-sheet-ios-group-margin-bottom - 2, null);
```
Comment (brandyscarney) [inline src/components/action-sheet/action-sheet.ios.scss]: Why do we need to use the mixin for `margin-bottom`, `margin-top`, `padding-bottom`, or `padding-top`? These shouldn't be affected by RTL, no?

File: src/components/split-pane/split-pane.scss [hunk 2]
```diff
@@ -20,10 +20,6 @@ $split-pane-side-max-width: 28% !default;
   contain: strict;
 }
 
-[dir="rtl"] .split-pane {
-  flex-direction: row-reverse;
```
Comment (brandyscarney) [inline src/components/split-pane/split-pane.scss]: Why was this removed?

File: src/themes/ionic.mixins.scss [hunk 3]
```diff
@@ -82,6 +82,15 @@
 
 // ---------------------------------------------------------------------------------
 
+// Minimum breakpoint width. Null for the smallest (first) breakpoint.
```
Comment (brandyscarney) [inline src/themes/ionic.mixins.scss]: Why were these moved?

File: src/themes/ionic.mixins.scss [hunk 4]
```diff
@@ -157,16 +153,44 @@
   }
 }
 
-$include-rtl: true !default;
+$app-direction: null !default;
+
+@mixin multi-dir() {
+  @if $app-direction == null {
+    $root: #{&};
+    @at-root [dir="ltr"], [dir="rtl"] {
```
Comment (brandyscarney) [inline src/themes/ionic.mixins.scss]: Why do we need to specify `dir` if the rule applies to both? Also will this work without `@at-root` being in front of `[dir="rtl"]`?

File: src/themes/ionic.mixins.scss [hunk 5]
```diff
@@ -157,16 +153,44 @@
...
     }
+  } @else if $app-direction == rtl {
+    @content;
+  }
+}
+
+@mixin ltr() {
+  @if $app-direction == null {
+    $root: #{&};
+    @at-root [dir="ltr"] {
```
Comment (brandyscarney) [inline src/themes/ionic.mixins.scss]: I'm not crazy about forcing a `dir` in the app.

File: src/components/action-sheet/action-sheet.ios.scss (hunk 1 above)
Comment (AmitMY) [inline src/components/action-sheet/action-sheet.ios.scss]: In theory, you are correct, and that is what I've done before. In practice no.
```scss
span {
    margin-top: 1px;
}
[dir=rtl] span {
    margin: 5px;
}
```

Gives the span `margin: 5px`, because of specificity.

```scss
[dir=rtl] span {
    margin-top: 1px;
}
[dir=rtl] span {
    margin: 5px;
}
```

Gives the span `margin: 1px 5px 5px 5px`, because of specificity. (which is the wanted behavior)

Don't worry, it gets super squashed at the end

File: src/components/split-pane/split-pane.scss (hunk 2 above)
Comment (AmitMY) [inline src/components/split-pane/split-pane.scss]: because it is not the correct behavior.. I checked on an iPad, in the mail app, the menu part is on the right on rtl, and left on ltr

File: src/themes/ionic.mixins.scss (hunk 3 above)
Comment (AmitMY) [inline src/themes/ionic.mixins.scss]: Probably incorrect to do this in this PR, but the order of the mixins is giving IDE errors, as we call an undefined method

File: src/themes/ionic.mixins.scss (hunk 4 above)
Comment (AmitMY) [inline src/themes/ionic.mixins.scss]: Because of CSS specificity rules, otherwise it overrides stuff. And no, the `@at-root` seems to be a must
